pkg/reconciler/labeler/v2/accessors.go

```diff
@@ -182,31 +197,45 @@ func NewConfigurationAccessor(
 }
 
 // list implements Accessor
-func (c *Configuration) list(ns, name string) ([]kmeta.Accessor, error) {
-	cl, err := c.configurationLister.Configurations(ns).List(labels.SelectorFromSet(labels.Set{
-		serving.RouteLabelKey: name,
-	}))
+func (c *Configuration) list(ns, routeName string, state v1.RoutingState) ([]kmeta.Accessor, error) {
+	cl, err := c.configurationLister.Configurations(ns).List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 	// Need a copy to change types in Go
-	kl := make([]kmeta.Accessor, len(cl))
-	for i, c := range cl {
-		kl[i] = c
+	kl := make([]kmeta.Accessor, 0, len(cl))
+	for _, c := range cl {
+		if _, has := getListAnnValue(c.Annotations, serving.RoutesAnnotationKey, routeName); has {
+			kl = append(kl, c)
+		}
 	}
 	return kl, err
 }
 
+func getListAnnValue(annotations map[string]string, key, value string) ([]string, bool) {
+	l, ok := annotations[key]
+	if ok {
```

Comment:
I think you can just drop this check. Empty string split will return an empty list

Reply:
It doesn't! String split instead returns an array of length 1 with an empty string. I've moved the check, but let's keep it in

Reply:
:/
not cool for `Split`, In any case I think we're better off with sets.String here.

Reply:
Sets requires yet another structure, but makes the remove easy, but then makes the string.join(str, ",") annoying. I think we need to write a helper either way (either for the remove or the join)

Reply:
nvm there's a nice to list function in set anyway. Changed it over